lib/Sema/TypeCheckDecl.cpp

```diff
@@ -4372,6 +4372,8 @@ class DeclChecker : public DeclVisitor<DeclChecker> {
 
     TC.checkDeclAttributes(PD);
 
+    PD->computeRequirementSignature();
```

Comment:
Hmm... should this go into `TypeChecker::validateDecl()`? It seems like we'd want the requirement signature most of the time that we look at the `ProtocolDecl`.